stdlib/public/stubs/SwiftNativeNSXXXBase.mm.gyb

```diff
@@ -95,6 +95,17 @@ SWIFT_RUNTIME_STDLIB_API
 - (BOOL)retainWeakReference {
   return swift_tryRetain(reinterpret_cast<HeapObject*>(self)) != nullptr;
 }
+- (void)_setWeaklyReferenced {
+  auto heapObj = reinterpret_cast<HeapObject *>(self);
+  heapObj->refCounts.setPureSwiftDeallocation(false);
+}
+- (bool)_setAssociatedObject:(id)obj
+                      forKey:(const void *)key
+           associationPolicy:(objc_AssociationPolicy)policy {
+  auto heapObj = reinterpret_cast<HeapObject *>(self);
+  heapObj->refCounts.setPureSwiftDeallocation(false);
+  return false; //let libobjc know that we didn't take care of everything
+}
```

Comment:
Can we consolidate this with the copy in `SwiftObject.mm`? Maybe make it a `#define`? Barring that, it would be good a comment in both places reminding the reader that these two implementations need to stay in sync.